shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -268,13 +283,23 @@ private void setSystemChromeEnabledSystemUIMode(PlatformChannel.SystemUiMode sys
       // the system overlay has changed. The overlays cannot be dismissed, so adding callback
       // support will allow users to restore the system ui and dismiss the overlays.
       // Not compatible with top/bottom overlays enabled.
-      enabledOverlays =
-          View.SYSTEM_UI_FLAG_IMMERSIVE
-              | View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-              | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-              | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-              | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-              | View.SYSTEM_UI_FLAG_FULLSCREEN;
+      if (Build.VERSION.SDK_INT >= 30) {
+        windowInsetsControllerCompat.setSystemBarsBehavior(
+            WindowInsetsControllerCompat.BEHAVIOR_SHOW_BARS_BY_SWIPE);
+        window.setDecorFitsSystemWindows(false);
+        windowInsetsControllerCompat.hide(WindowInsets.Type.statusBars());
+        windowInsetsControllerCompat.hide(WindowInsets.Type.navigationBars());
+
+        return;
```

Comment:
is this return intentional?

Reply:
Yes because I don't want the default behavior to run (line 347) if any of these cases are hit in theory. There's a return after each case is handled if the SDK version is >= 30